hack/update-bazel.sh

```diff
@@ -18,6 +18,8 @@ set -o nounset
 set -o pipefail
 
 TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
+# https://github.com/kubernetes/test-infra/issues/5699#issuecomment-348350792
+cd $TESTINFRA_ROOT
```

Comment:
nit: `cd ${TESTINFRA_ROOT}`
The other way to do it that's more consistent with k/k hack/ would be using pushd/popd.
/shrug ?

Reply:
maybe, but we already rely on this heavily throughout the script

Reply:
We don't need to though. This could be done very similarly to how https://github.com/kubernetes/kubernetes/blob/master/hack/update-bazel.sh is done. Perhaps going with the bash source method similar to k/k would solve this completely?

Reply:
the style nit applies, otherwise this LGTM.

(FWIW I think k/k's `hack/update-bazel.sh` also doesn't work properly when it's called outside of the repo root.)

Reply:
fixed style nit, I also think we should probably continue to get it this way.